gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -0,0 +1,790 @@
+                lambda x: x.drop(
+                    "sift_prediction",
+                    "sift_score",
+                    "polyphen_prediction",
+                    "polyphen_score",
+                )
+            )
+        )
+    }
+    return selects
+
+
+def get_select_global_fields(ht: hl.Table) -> Dict[str, hl.expr.Expression]:
+    """
+    Generate a dictionary of globals to select by checking the config of all tables joined.
+
+    :param ht: Final joined HT with globals.
+    """
+    t_globals = []
+    for t in TABLES_FOR_RELEASE:
+        config = get_config().get(t)
+        if "select_globals" in config:
+            select_globals = get_select_fields(config["select_globals"], ht)
+            if "global_name" in config:
+                global_name = config.get("global_name")
+                select_globals = {global_name: hl.struct(**select_globals)}
+            t_globals.append(select_globals)
+
+    t_globals = reduce(lambda a, b: dict(a, **b), t_globals)
+
+    return t_globals
+
+
+def get_select_fields(
+    selects: Union[List, Dict], base_ht: hl.Table
+) -> Dict[str, hl.expr.Expression]:
+    """
+    Generate a select dict from traversing the `base_ht` and extracting annotations.
+
+    :param selects: Mapping or list of selections.
+    :param base_ht: Base Hail Table to traverse.
+    :return: select Mapping from annotation name to `base_ht` annotation.
+    """
+    select_fields = {}
+    if selects is not None:
+        if isinstance(selects, list):
+            select_fields = {selection: base_ht[selection] for selection in selects}
+        elif isinstance(selects, dict):
+            for key, val in selects.items():
+                # Grab the field and continually select it from the hail table.
+                ht = base_ht
+                for attr in val.split("."):
+                    ht = ht[attr]
+                select_fields[key] = ht
+    return select_fields
+
+
+def get_ht(
+    dataset: str,
+    _intervals: nullable(sequenceof(anytype)),
+    test: bool,
+    release_exists: bool,
+) -> hl.Table:
+    """
+    Return the appropriate Hail table with selects applied.
+
+    :param dataset: Hail Table to join.
+    :param _intervals: Intervals for reading in hail Table. Used to optimize join.
+    :param test: Whether call is for a test run.
+    :param release_exists: Whether the release HT already exists.
+    :return: Hail Table with fields to select.
+    """
+    logger.info("Getting the %s dataset and its selected annotations...", dataset)
+    config = get_config(release_exists=release_exists)[dataset]
+
+    # There is no single path for insilico predictors, so we need to handle this case
+    # separately.
+    if dataset == "in_silico":
+        base_ht = config["ht"]
+    else:
+        ht_path = config["path"]
+        logger.info("Reading in %s", dataset)
+        base_ht = hl.read_table(ht_path, _intervals=_intervals)
+        if dataset == "freq":
```

Comment:
Again, if the difference between the datasets are this small, we can just add a data_type parameter and use the function for both